mantle/kola/tests/rhcos/upgrade.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rhcos
+
+import (
+	"path/filepath"
+
+	"github.com/coreos/pkg/capnslog"
+
+	"github.com/coreos/mantle/kola"
+	"github.com/coreos/mantle/kola/cluster"
+	"github.com/coreos/mantle/kola/register"
+	"github.com/coreos/mantle/kola/tests/util"
+	"github.com/coreos/mantle/platform/conf"
+)
+
+var plog = capnslog.NewPackageLogger("github.com/coreos/mantle", "kola/tests/upgrade")
+
+func init() {
+	register.RegisterUpgradeTest(&register.Test{
+		Run:         rhcosUpgradeLuks,
+		ClusterSize: 1,
+		// if renaming this, also rename the command in kolet-httpd.service below
+		Name:     "rhcos.upgrade.luks",
+		FailFast: true,
+		Tags:     []string{"upgrade"},
```

Comment:
All the tests in `run-upgrade` are a separate set from the `run` ones. They don't run together in the same invocation. And of course, all the tests in `run-upgrade` are understood to be about upgrades. :)

Reply:
I just cargo culted this from the fcos one.

Reply:
Ack, looks like it came in with https://github.com/coreos/coreos-assembler/pull/1383. Not a big deal.